### ISSUE TITLE
Monitor the resource over HTTP(s) proxy

### ISSUE THREAD

**Issue by ugurerkan** [louislam/uptime-kuma#470]
**Is your feature request related to a problem?**
When need to monitor a resource over a different or private network than the server, we are not able to proxy at the monitoring resource level.

**Describe the solution you'd like**

Simply, proxy support of axios we could provide this functionality. With proper configuration steps at the monitor resource level, we get proxy configuration from the user and implement recorded proxy config for [HTTP(s) requests](https://github.com/louislam/uptime-kuma/blob/1ebf752f1a6e37a8bdeb54458a8958488a46af52/server/model/monitor.js#L138).

Related request section
```js
// server/model/monitor.js#L13
let res = await axios.get(this.url, {
    timeout: this.interval * 1000 * 0.8,
    headers: {
        "Accept": "*/*",
        "User-Agent": "Uptime-Kuma/" + version,
    },
    httpsAgent: new https.Agent({
        maxCachedSessions: 0,      // Use Custom agent to disable session reuse (https://github.com/nodejs/node/issues/3940)
        rejectUnauthorized: ! this.getIgnoreTls(),
    }),
    maxRedirects: this.maxredirects,
    validateStatus: (status) => {
        return checkStatusCode(status, this.getAcceptedStatuscodes());
    },
});
```

Related [axios request config](https://axios-http.com/docs/req_config)
```js
  // `proxy` defines the hostname, port, and protocol of the proxy server.
  // You can also define your proxy using the conventional `http_proxy` and
  // `https_proxy` environment variables. If you are using environment variables
  // for your proxy configuration, you can also define a `no_proxy` environment
  // variable as a comma-separated list of domains that should not be proxied.
  // Use `false` to disable proxies, ignoring environment variables.
  // `auth` indicates that HTTP Basic auth should be used to connect to the proxy, and
  // supplies credentials.
  // This will set an `Proxy-Authorization` header, overwriting any existing
  // `Proxy-Authorization` custom headers you have set using `headers`.
  // If the proxy server uses HTTPS, then you must set the protocol to `https`. 
  proxy: {
    protocol: 'https',
    host: '127.0.0.1',
    port: 9000,
    auth: {
      username: 'mikeymike',
      password: 'rapunz3l'
    }
  },
```

**Describe alternatives you've considered**
IMO, this function should provide in uptime-kuma and at the monitor resource level. Also, if all monitor resources should work on a different network, a system-wide proxy or VPN could be an alternative option.

---

If this feature is approved for development, I could create a PR for implement 🚀

**Comment by louislam** [comment]
It is likely to be many-to-one. 
  - one monitor has one proxy config
  - one proxy config can be used in multiple monitors

**Comment by ugurerkan** [comment]
Thank you for your review @louislam 
Agree, it is will be better to design like this. If you think this is a valuable addition to the uptime kuma, I can create a draft PR then start to develop this feature. So we can review and fine-tune the feature from PR to get it better.

**Comment by louislam** [comment]
Welcome! 

But I suggest that you should test the `proxy` property first before implementing the feature, as it may conflict with `httpsAgent`.

**Comment by ugurerkan** [comment]
Thank you very much, I'll double check on this 👮🏻‍♂️

**Comment by ugurerkan** [comment]
Hey @louislam,

I have created a PR for proxy support. Like your foresee, I have encountered issues when using proxy directly with axios, added new HTTP and HTTPS agents as a solution from [proxy-agent ](https://www.npmjs.com/package/proxy-agent) library. They allow us to use proxied HTTP/S agents and solves connection issues with proxied HTTPS requests.

I have tested proxied HTTP and HTTPS request and proxy management operations like create, delete, export, import, etc they are working like expected.

Kindly request your reviews  🚀
